crates/iota-types/src/transaction.rs

```diff
@@ -2342,6 +2370,23 @@ impl SenderSignedData {
 
         Ok(tx_size)
     }
+
+    // TODO: A temporary created function. Needs to be replaced with a proper check.
+    pub fn move_authenticator(&self) -> Option<&MoveAuthenticator> {
+        let signatures = self.tx_signatures();
+
+        if signatures.len() == 1 && signatures[0].is_move_authenticator() {
+            match &signatures[0] {
```

Comment:
Are the other cases important? It doesn't seems so, you could replace
the whole `match` expression with a 
            
```suggestion
if let GenericSignature::MoveAuthenticator(move_authenticator) = &signatures[0] {
    Some(move_authenticator)
} else {
    None
}
```

In any case the unreacheable is suspect. From the authority I see it being called without any additional checks.

Reply:
I didn't pay a lot of attention here because I supposed, this function is temporary and will be removed.
But I like your option, it fixed here: 6132898210